### PR TITLE
searchjobs: hook up graphql CreateSearchJob and NodeResolvers

### PR DESCRIPTION
**Pull request by keegancsmith**
This required fleshing out the service layer a bunch. We don't really do anything yet other than put an entry for search job into the table and can fetch it again.

Additionally I added in the security we discussed where only a user can see there own search jobs.

Store tests for Get are still missing. Right now our codebase contains a mix of the word exhaustive and search jobs. But that should be invisible to the user, so renaming types will be great when there isn't a bunch of inflight work.

Test Plan: Wrote unit tests. Also did manual testing with the following graphql queries

```graphql
mutation {
  createSearchJob(query: "testing testing 123") {
    id
    query
  }
}

query {
  node(id: "U2VhcmNoSm9iOjE=") {
    ...on SearchJob {
      query
    }
  }
}
```

### CODE REVIEW

File: internal/search/exhaustive/service/service.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return tx.GetExhaustiveSearchJob(ctx, jobID)
+}
+
+func (s *Service) GetSearchJob(ctx context.Context, id int64) (_ *types.ExhaustiveSearchJob, err error) {
+	ctx, _, endObservation := s.operations.createSearchJob.With(ctx, &err, opAttrs(
```
Comment (stefanhengl) [inline internal/search/exhaustive/service/service.go:111]:
```suggestion
	ctx, _, endObservation := s.operations.getSearchJob.With(ctx, &err, opAttrs(
```

File: internal/search/exhaustive/service/service.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	tx, err := s.store.Transact(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { err = tx.Done(err) }()
+
+	// XXX(keegancsmith) this API for creating seems easy to mess up since the
+	// ExhaustiveSearchJob type has lots of fields, but reading the store
+	// implementation only two fields are read.
```
Comment (stefanhengl) [inline internal/search/exhaustive/service/service.go:98]: Shall we use a separate type for creating the job?

File: internal/search/exhaustive/store/exhaustive_search_jobs_test.go [hunk 3]
```diff
@@ -64,10 +71,12 @@ func TestStore_CreateExhaustiveSearchJob(t *testing.T) {
 			},
 			expectedErr: errors.New("missing query"),
 		},
+
+		// TODO(keegancsmith) for some reason we don't let users recreate searches.
```
Comment (stefanhengl) [inline internal/search/exhaustive/store/exhaustive_search_jobs_test.go:75]: This is surprising, isn't it? I thought we want user's to be able to rerun a job.

**Review by stefanhengl — APPROVED** [review]

File: internal/search/exhaustive/store/exhaustive_search_jobs_test.go (hunk 3 above)
Comment (keegancsmith) [reply internal/search/exhaustive/store/exhaustive_search_jobs_test.go:75]: agreed, we have a unique constraint we should follow up with and remove.

File: internal/search/exhaustive/service/service.go (hunk 2 above)
Comment (keegancsmith) [reply internal/search/exhaustive/service/service.go:98]: I think so. I think as more code is written we will have a nice shape between responsibilities of resolver, backend and store. Right now the backend part is quite thin.

**Pull request merged**